code/atom.dm

```diff
@@ -542,12 +554,26 @@
 		return
 
 	if (isturf(last_turf))
+		if(src.bound_width == 32 && src.bound_width == src.bound_height)
+			last_turf.passability_cache = null
+			last_turf.pass_unstable -= src.pass_unstable
+		else // This is ugly but ~30% faster than just going over old_locs
+			for(var/turf/covered_turf as anything in old_locs)
+				covered_turf.pass_unstable -= src.pass_unstable
+				covered_turf.passability_cache = null
 		if (src.event_handler_flags & USE_PROXIMITY)
 			last_turf.checkinghasproximity = max(last_turf.checkinghasproximity-1, 0)
 			for (var/turf/T2 in range(1, last_turf))
 				T2.neighcheckinghasproximity--
 	if(isturf(src.loc))
 		var/turf/T = src.loc
+		if(src.bound_width == 32 && src.bound_width == src.bound_height)
+			T.passability_cache = null
+			T.pass_unstable += src.pass_unstable
+		else
+			for(var/turf/covered_turf as anything in src.locs)
+				covered_turf.pass_unstable += src.pass_unstable
+				covered_turf.passability_cache = null
```

Comment:
I know you were talking about this earlier, but can't you just make it a macro that takes `T` as an argument?

Reply:
:whelm:
potentially, but would be one of the cursed ones.